metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,127 @@
+package elasticsearch
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/elastic/beats/metricbeat/helper"
+)
+
+// Global clusterIdCache. Assumption is that the same node id never can belong to a different cluster id
+var clusterIdCache = map[string]string{}
+
+type infoStruct struct {
+	ClusterName string `json:"cluster_name"`
+	ClusterId   string `json:"cluster_uuid"`
+}
+
+func GetClusterId(http *helper.HTTP, uri string, nodeId string) (string, error) {
+
+	// Check if cluster id already cached. If yes, return it.
+	if clusterId, ok := clusterIdCache[nodeId]; ok {
+		return clusterId, nil
+	}
+
+	// Makes sure the http uri is reset to its inital value
+	defer http.SetURI(uri)
+
+	info, err := getInfo(http, uri)
+	if err != nil {
+		return "", err
+	}
+
+	clusterIdCache[nodeId] = info.ClusterId
+	return info.ClusterId, nil
+}
+
+// IsMaster checks if the given node host is a master node
+//
+// The detection of the master is done in two steps:
+// * Fetch node name from /_nodes/_local/name
+// * Fetch current master name from cluster state /_cluster/state/master_node
+//
+// The two names are compared
+func IsMaster(http *helper.HTTP, uri string) (bool, error) {
+	// Makes sure the http uri is reset to its inital value
+	defer http.SetURI(uri)
+
+	node, err := getNodeName(http, uri)
+	if err != nil {
+		return false, err
+	}
+
+	master, err := getMasterName(http, uri)
+	if err != nil {
+		return false, err
+	}
+
+	return master == node, nil
+}
+
+func getNodeName(http *helper.HTTP, uri string) (string, error) {
+
+	// Parses the uri to replace the path
```

Comment:
Lines 64-73 are the same for the next 2 methods except for `u.Path`. Perhaps we can have a helper method that avoids repeating it all?

Reply:
done